### PR TITLE
Handle permission

### PR DESCRIPTION
**Pull request by trautamaki**

### CODE REVIEW

File: app/src/main/java/org/lineageos/tv/launcher/MainActivity.kt [hunk 1]
```diff
@@ -103,6 +103,11 @@ class MainActivity : FragmentActivity(R.layout.activity_main) {
         registerReceiver(PackageReceiver(), intentFilter)
     }
 
+    override fun onDestroy() {
+        super.onDestroy()
+        unregisterReceiver(PackageReceiver())
```
Comment (SebaUbuntu) [inline app/src/main/java/org/lineageos/tv/launcher/MainActivity.kt:108]: You're unregistering a new PackageReceiver object, it won't unregister the registered one

**Comment by SebaUbuntu** [comment]
Sorry, had to remove the commits related to permissions handling, I'll redo it with contracts and PermissionsUtils like Aperture and Glimpse

File: app/src/main/java/org/lineageos/tv/launcher/MainActivity.kt [hunk 2]
```diff
@@ -32,7 +32,6 @@ import org.lineageos.tv.launcher.utils.Suggestions.orderSuggestions
 class MainActivity : FragmentActivity(R.layout.activity_main) {
     // Views
     private val mainVerticalGridView by lazy { findViewById<VerticalGridView>(R.id.main_vertical_grid) }
-    private val settingButton by lazy { findViewById<ImageButton>(R.id.settings_button) }
```
Comment (SebaUbuntu) [inline app/src/main/java/org/lineageos/tv/launcher/MainActivity.kt:35]: Eh, we usually leave views there even if used only once, like view bindings

File: app/src/main/java/org/lineageos/tv/launcher/adapter/MainVerticalAdapter.kt [hunk 3]
```diff
@@ -86,4 +86,10 @@ class MainVerticalAdapter(
         Collections.swap(rowList, from, to)
         notifyItemMoved(from, to)
     }
+
+    fun addItems(mainItems: MutableList<Pair<Long, org.lineageos.tv.launcher.model.MainRowItem>>) {
```
Comment (luca020400) [inline app/src/main/java/org/lineageos/tv/launcher/adapter/MainVerticalAdapter.kt:90]: import?

File: app/src/main/java/org/lineageos/tv/launcher/MainActivity.kt [hunk 4]
```diff
@@ -41,21 +40,47 @@ class MainActivity : FragmentActivity(R.layout.activity_main) {
...
+    private val packageReceiver by lazy { PackageReceiver() }
+
     @SuppressLint("RestrictedApi")
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        if (checkCallingOrSelfPermission(TvContractCompat.PERMISSION_READ_TV_LISTINGS) != PackageManager.PERMISSION_GRANTED) {
-            requestPermissions(arrayOf(TvContractCompat.PERMISSION_READ_TV_LISTINGS), 0)
+        // Get items that don't need special permissions
+        val mainItems = ArrayList<Pair<Long, MainRowItem>>()
```
Comment (luca020400) [inline app/src/main/java/org/lineageos/tv/launcher/MainActivity.kt:50]: prefer using arrayListOf<T>()

File: app/src/main/java/org/lineageos/tv/launcher/MainActivity.kt (hunk 2 above)
Comment (luca020400) [reply app/src/main/java/org/lineageos/tv/launcher/MainActivity.kt:35]: Yeah keep that here, it's fine

**Review by luca020400 — APPROVED** [review]

**Pull request merged**